packages/stark-ui/src/modules/progress-indicator/reducers/progress-indicator.reducer.ts

```diff
@@ -0,0 +1,96 @@
+import { StarkProgressIndicatorConfig } from "../entities";
+import { StarkProgressIndicatorActions, StarkProgressIndicatorActionTypes } from "../actions/progress-indicator.actions";
+
+/**
+ * @ignore
+ */
+const _cloneDeep: Function = require("lodash/cloneDeep");
+
+/**
+ * Initial state of the reducer
+ */
+const INITIAL_STATE: Map<string, StarkProgressIndicatorConfig> = new Map<string, StarkProgressIndicatorConfig>();
+
+/**
+ * Reducer for the progress indicator
+ * @param Map<string, StarkProgressIndicatorConfig> the actual state of the progress indicator
+ * @param StarkProgressIndicatorActions the action to perform
+ * @returns Map<string, StarkProgressIndicatorConfig> the new state of the progress inidcator
+ */
+//tslint:disable:cognitive-complexity
```

Comment:
If this tslint flag is needed, could you add a TODO to refactor this function to reduce the cognitive complewity?